### PR TITLE
Added continuous integration.

### PR DESCRIPTION
**Pull request by pelson**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+install:
+  # Fetch and install conda
+  # -----------------------
+  - export CONDA_BASE=http://repo.continuum.io/miniconda/Miniconda3
+  - wget ${CONDA_BASE}-latest-Linux-x86_64.sh -O miniconda.sh;
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - export PATH="$HOME/miniconda/bin:$PATH"
+  - conda config --set show_channel_urls True
+  - conda config --add channels scitools
```
Comment (rhattersley) [inline .travis.yml]: Remove?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  - rm -rf iris_source
+
+  # Summarise the environment
+  # -------------------------
+  - conda list
+
+  # Install iris-grib itself
+  # ------------------------
+  - pip install . --no-deps
+  - python -c 'import iris; print(iris); print(iris.__version__)'
```
Comment (rhattersley) [inline .travis.yml]: Remove

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - export PATH="$HOME/miniconda/bin:$PATH"
+  - conda config --set show_channel_urls True
+  - conda config --add channels scitools
+  - conda config --add channels conda-forge
+
+  # Install iris-grib's dependencies
+  # --------------------------------
+  - export IRIS_DEPS="biggus numpy cf_units setuptools pyke netcdf4 scipy cartopy"
+  - conda install --quiet --yes python=${python} pip nomkl ${IRIS_DEPS} python-ecmwf_grib=${grib_api}
```
Comment (rhattersley) [inline .travis.yml]: Let's install iris (+ hence its dependencies) with conda, with an option to uninstall and install from source instead.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            iris=${iris}* \
+            python-ecmwf_grib=${grib_api}*
+
+  # Fetch and install Iris from source
+  # ----------------------------------
+  - |
+     if [ -n "$iris_from_source" ];
+        conda remove --yes iris
+        wget https://github.com/scitools/iris/archive/${iris}.tar.gz -O iris.tar.gz
+        tar xzf iris.tar.gz && rm -rf iris.tar.gz && mv iris-${iris} iris_source
```
Comment (rhattersley) [inline .travis.yml]: These two `{iris}` substitutions should be `{iris_from_source}`.

**Comment by rhattersley** [comment]
Not sure why travis is failing:
```
$ conda install --quiet --yes python=${python} pip nomkl \ iris=${iris}* \ python-ecmwf_grib=${grib_api}*
Error: Invalid package specification:  iris=1.9*
```
and
```
$ conda install --quiet --yes python=${python} pip nomkl \ iris=${iris}* \ python-ecmwf_grib=${grib_api}*
Error: Invalid package specification:  iris=*
```
I've tried doing the same thing on my machine (conda 4.0.5) and it works fine. :disappointed:

**Comment by pelson** [comment]
> I've tried doing the "same" thing on my machine (conda 4.0.5) and it works fine. :disappointed:

Likewise when developing. I can only assume it is a travis-ism, and I have made it a proper multi-line statement within `.travis.yml`.

**Comment by pelson** [comment]
Fails because `iris_grib` doesn't work with v1.9 currently (being addressed elsewhere).

**Comment by rhattersley** [comment]
> Fails because iris_grib doesn't work with v1.9 currently (being addressed elsewhere).

:+1:

**Pull request merged**